packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -2326,14 +2326,47 @@ class WebpackCLI {
       return;
     }
 
+    const isCacheEnabled = compiler.compilers
+      ? compiler.compilers.some((compiler) => compiler.options.cache)
+      : compiler.options.cache;
+
+    let needForceShutdown = false;
+
+    if (isCacheEnabled) {
+      const signals = ["SIGINT", "SIGTERM"];
+
+      signals.forEach((signal) => {
+        const listener = () => {
+          if (needForceShutdown) {
+            process.exit();
+          }
+
+          this.logger.info(
+            "Gracefully shutting down. To force exit, press ^C again. Please wait...",
+          );
+
+          needForceShutdown = true;
+
+          compiler.close(() => {
+            process.exit();
+          });
+        };
+
+        process.on(signal, listener);
+      });
+    }
+
     const isWatch = (compiler) =>
       compiler.compilers
         ? compiler.compilers.some((compiler) => compiler.options.watch)
         : compiler.options.watch;
 
     if (isWatch(compiler) && this.needWatchStdin(compiler)) {
       process.stdin.on("end", () => {
-        process.exit(0);
+        compiler.close(() => {
```

Comment:
It can be problem for webpack v4, because no `close` method, we should check it `if (typeof compiler.close === "function")`

Reply:
Okay on it 👍🏽

Reply:
Done